### PR TITLE
quick tidy

### PR DESCRIPTION
**Pull request by oliver-sanders**
Built on #201 

* Merge the two different plugins sections I had created.
* Merge the three task-job sections under a single section.
* Add "this is beta software" warnings to the docs.

Structure a bit simpler now:

Screenshot from 2021-03-15 11-32-03

### CODE REVIEW

File: src/_templates/main_loop_plugin.rst [hunk 1]
```diff
@@ -7,6 +7,10 @@
       otherwise you can end up with imported modules and mess like that
 
    {% block functions %}
+
+   Python API
+   ----------
+
```
Comment (oliver-sanders) [inline src/_templates/main_loop_plugin.rst:13]: This header separates the API docs in the main-loop plugins sections from the rest of the content as it wasn't really clear what they meant before.

File: src/plugins/index.rst [hunk 2]
```diff
@@ -9,3 +9,9 @@ The following are "core" plugins maintained by the Cylc team:
    :maxdepth: 1
 
    cylc-rose
+   main-loop/index
+
+.. warning::
+
+   Plugins are a new feature of Cylc8 and the interface is likely to change in
+   future releases.
```
Comment (oliver-sanders) [inline src/plugins/index.rst:17]: https://github.com/cylc/cylc-flow/issues/4106

File: src/index.rst [hunk 3]
```diff
@@ -5,6 +5,12 @@ Cylc ("*silk*") is a workflow engine for cycling systems - it orchestrates
 distributed suites of interdependent cycling tasks that may continue to run
 indefinitely.
 
+.. warning::
+
+   Cylc8 is beta software that is not yet recommended for production use.
```
Comment (hjoliver) [inline src/index.rst:10]:
```suggestion
   Cylc 8 is beta software that is not yet recommended for production use.
```

File: src/plugins/index.rst [hunk 4]
```diff
@@ -9,3 +9,9 @@ The following are "core" plugins maintained by the Cylc team:
    :maxdepth: 1
 
    cylc-rose
+   main-loop/index
+
+.. warning::
+
+   Plugins are a new feature of Cylc8 and the interface is likely to change in
```
Comment (hjoliver) [inline src/plugins/index.rst:16]:
```suggestion
   Plugins are a new feature of Cylc 8 and the interface is likely to change in
```

File: src/tutorial/introduction.rst [hunk 5]
```diff
@@ -3,6 +3,12 @@
 Introduction
 ============
 
+.. TODO
+
+.. warning::
+
+   This section has not yet been upgraded to match Cylc8 changes.
```
Comment (hjoliver) [inline src/tutorial/introduction.rst:10]:
```suggestion
   This section has not yet been upgraded to match Cylc 8 changes.
```

File: src/tutorial/runtime/index.rst [hunk 6]
```diff
@@ -3,6 +3,12 @@
 Runtime
 =======
 
+.. TODO
+
+.. warning::
+
+   This section has not yet been upgraded to match Cylc8 changes.
```
Comment (hjoliver) [inline src/tutorial/runtime/index.rst:10]:
```suggestion
   This section has not yet been upgraded to match Cylc 8 changes.
```

File: src/tutorial/scheduling/index.rst [hunk 7]
```diff
@@ -3,6 +3,12 @@
 Scheduling
 ==========
 
+.. TODO
+
+.. warning::
+
+   This section has not yet been upgraded to match Cylc8 changes.
```
Comment (hjoliver) [inline src/tutorial/scheduling/index.rst:10]:
```suggestion
   This section has not yet been upgraded to match Cylc 8 changes.
```

File: src/user-guide/task-implementation/job-scripts.rst [hunk 8]
```diff
@@ -249,3 +249,23 @@ task to repeatedly poll for the results of the detached processes:
...
+.. _PipeInJobScripts:
+
+Use of pipes in job scripts
+---------------------------
+
+In bash, the return status of a pipeline is normally the exit status of the
+last command. This is unsafe, because if any command in the pipeline fails, the
+script will continue nevertheless.
+
+For safety, a cylc task job script running in bash will have the
```
Comment (hjoliver) [inline src/user-guide/task-implementation/job-scripts.rst:263]:
```suggestion
For safety, a Cylc task job script running in bash will have the
```

File: src/tutorial/index.rst [hunk 9]
```diff
@@ -3,6 +3,12 @@
 Tutorial
 ========
 
+.. TODO
+
+.. warning::
+
+   This section has not yet been upgraded to match Cylc8 changes.
```
Comment (hjoliver) [inline src/tutorial/index.rst:10]:
```suggestion
   This section has not yet been upgraded to match Cylc 8 changes.
```

**Review by hjoliver — APPROVED** [review]
Good, thanks @oliver-sanders

File: src/user-guide/task-implementation/job-scripts.rst [hunk 10]
```diff
@@ -249,3 +249,23 @@ task to repeatedly poll for the results of the detached processes:
            # job.done indicator file does not exist yet.
            script = "[[ ! -f $RUN_DIR/job.done ]] && exit 1"
            execution retry delays = 10 * PT1M
+
+
+.. _PipeInJobScripts:
+
+Use of pipes in job scripts
```
Comment (datamel) [inline src/user-guide/task-implementation/job-scripts.rst:256]:
```suggestion
Use of Pipes in Job Scripts
```

Case change to keep in line with other headings.

**Review by datamel — APPROVED** [review]
I'm happy with this change, I have read through the change and built the docs. I have one minor suggestion (case change).

File: src/user-guide/task-implementation/job-scripts.rst (hunk 10 above)
Comment (oliver-sanders) [reply src/user-guide/task-implementation/job-scripts.rst]: Applied.

**Pull request merged**